### PR TITLE
Added stack_map to opsworks.js.erb

### PR DESCRIPTION
**Pull request by alxshr**

### CODE REVIEW

**Comment by cbbarclay** [comment]
Thank you, we will review your pull request.

**Comment by githuesch** [comment]
Manually generating JSON is hard to follow. I would rather manipulate the data structure and then call `to_json` on the result. Furthermore, I don't see why this has to be a part of the file we generate. Feel free to use a custom recipe to generate another file and use that one.

**Comment by alxshr** [comment]
I agree with the part of manipulating data structure will be more efficient and easier to follow, but I do not agree with the part that it is not needed. I think it is needed in order to provide consistency in opsworks for diffrenet layers. The opsworks.php for PHP layer provides this information, but opsworks.js for NoideJS does not do so. Since we use both I expected the "out of the box" meta information to be consistent across layers. Please reconsider.

**Comment by githuesch** [comment]
You are right, I totally forgot about opsworks.php. Will now have a closer look, sorry.

**Comment by githuesch** [comment]
I think what tripped me up was the name `stack_map`. I didn't recall having anything like that in any public API part, and in fact, `stack_map` is declared private in opsworks.php. Then there are two public functions, `layers()` and `hosts(layer)`. From a consistency standpoint, the API should be the same across languages. On the other hand, in different languages different things are considered idiomatic code. I will discuss this internally.

**Comment by githuesch** [comment]
We implemented something more similar to what you see in opsworks.php. You will be able to use it as follows.

``` Ruby
var opsworks = require('opsworks');

var layers = opsworks.layers(); // will give you an array of layers
var hosts = opsworks.hosts(layers[0]); // will give you an array of IP addresses
```

I'm not going to merge your request, but I can update it once our change is rolled out.

**Comment by githuesch** [comment]
@alxshr This is rolled out now.
